src/sdk/namespace_fs.js

```diff
@@ -1042,6 +1051,10 @@ class NamespaceFS {
                 return content_dir_info;
             }
 
+            if (!await this._is_storage_class_supported(params.storage_class)) {
```

Comment:
you might want to consider changing the is_storage_class_supported() to throw_if_storage_class_not_supported() and throw the error from inside the function.